### ISSUE TITLE
Editor: Posts are opened in both the app and browser

### ISSUE THREAD

**Issue by rachelmcr** [Automattic/wp-desktop#697]
#### Steps to reproduce

I'm not sure the exact steps needed to consistently reproduce this issue. I had the app open for quite a while before I started seeing this happen, and after a restart I haven't seen it again (although I'll keep checking for initial steps to trigger this).

Once it started happening, these steps consistently reproduced it (on various sites):

1. Open the desktop app.
2. Go to My Sites > Posts.
3. Select a post to edit.

#### What I expected

I expect the post to open in the block editor in the app.

#### What happened instead

The post opened in the block editor in the app AND in a new browser tab.

#### App Version / OS version

WPDesktop 4.6.0-beta1 / macOS 10.15

#### Context / Source
manual-testing

**Comment by rachelmcr** [comment]
I don't think this is a new issue. I did some digging and I see we had a recent report of this in https://github.com/Automattic/wp-desktop/issues/691 (version 4.4.2) and both @designsimply and I saw it happen while testing version 4.1.0 (internal ref p5T066-Ok-p2#comment-2740). However, I'm still not sure what triggers it.

**Comment by rachelmcr** [comment]
I started seeing this issue again, but this time the block editor never finishes loading in the app; it only loads in the browser tab that opens. I'm also seeing this console error (with personal details redacted) when it happens:

```
https://MY_BLOG_SUBDOMAIN.wordpress.com/wp-login.php?redirect_to=https%3A%2F%2FMY_BLOG_SUBDOMAIN.wordpress.com%2Fwp-admin%2Fpost.php%3Fpost%3D9397%26action%3Dedit%26calypsoify%3D1%26block-editor%3D1%26frame-nonce%3DNONCE%26origin%3Dhttp%253A%252F%252F127.0.0.1%253A41050%26environment-id%3Ddesktop&reauth=1
Failed to set referrer policy: The value 'strict-origin-when-cross-origin' is not one of 'always', 'default', 'never', 'origin-when-crossorigin', 'no-referrer', 'no-referrer-when-downgrade', 'origin', 'origin-when-cross-origin', or 'unsafe-url'. The referrer policy has been left unchanged.
```

When I exit the editor in the browser tab, it takes me back to this page: http://127.0.0.1:41050/desktop/hey.html

If I restart the app it stops happening, and in that case I still see the `Failed to set referrer policy` console error when I open the block editor but it links straight to `https://MY_BLOG_SUBDOMAIN.wordpress.com/wp-admin/post.php...` without the `https://MY_BLOG_SUBDOMAIN.wordpress.com/wp-login.php?redirect_to=` redirect.

**Comment by rachelmcr** [comment]
cc @apeatling & @gwwar — I wanted to give you a heads up about this issue and see if you can gauge the priority for looking into it further.

**Comment by zdenys** [comment]
Another report in 2530128-zen for WordPress 4.6.0 on macOS.

**Comment by michaelh99** [comment]
Happening to me too.  Not sure what's triggering it.
I had some substantial changes to my Win10 machine and had to "reset windows" which take it back to some version of stock and uninstalls all apps.
Now that I've reinstalled the WP 4.6 app, every time I go to edit a post it opens the editor in Firefox (my default browser) rather than WP, though it does bring up a blank window in the app.

**Comment by michaelh99** [comment]
I tried uninstalling the app and reinstalling and also enabled the classic editor plugin.  After reinstalling, posts are opening in the app editor.  I disabled the classic editor plugin and posts started opening up in my browser.  Re-enabling it and posts are only opening in the app.
So, I'm not sure what might be causing the new block editor to confuse the app, but for now I'll use the old editor until someone has some other suggestion

**Comment by rachelmcr** [comment]
I just came across https://github.com/Automattic/wp-calypso/pull/36977 and the timing of this change could explain the rise in this issue in the app. From that PR:

> If load of editor into iframe fails due to cross origin issues, eg. if a Jetpack site is setting X-Frame-Options to same-origin at the web server level, then redirect editor to wp-admin

@glendaviesnz Do you think that change could have caused the issue described here in the desktop app?

**Comment by glendaviesnz** [comment]
> Do you think that change could have caused the issue described here in the desktop app?

@rachelmcr sorry about the delay in replying. That is possible, but should only affect jetpack or atomic sites. Can you confirm if when you experienced the issue on the desktop app it was a jetpack/atomic site or a simple site? The 'MY_BLOG_SUBDOMAIN.wordpress.com' in the details above seems to indicate it was a simple site.

**Comment by rachelmcr** [comment]
@glendaviesnz It was a simple site in my case, so this app issue isn't limited  to jetpack/atomic sites. Curiouser and curiouser.

**Comment by glendaviesnz** [comment]
Thanks for clarifying @rachelmcr, there is likely to be a bit of work getting done early in the new year around cross origin things, as Chrome is tightening up how they handle this, will leave it to @gwwar to decide if this gets looked at as part of this work.

**Comment by ajaykj** [comment]
Another report in 17710220-hc today. The user was using the Desktop app for Windows. Follow-up with the user here: 2607589-zen

**Comment by nsakaimbo** [comment]
So I _think_ I've identified a set of steps that seem to reproduce this issue fairly consistently and opened an issue against Calypso.

Ref: Automattic/wp-calypso/issues/40376

**Comment by nsakaimbo** [comment]
Update on this thread: we're closing this issue as we expect this and a whole host of other issues to be resolved by the new version of the application (v7.0.5) that was just released. The latest build can now be downloaded at the WordPress Desktop landing page [here](P7jrAc-3-p2).
